File: app/src/main/java/com/example/mike/mp3player/client/callbacks/TrackViewPagerChangeCallback.java
package com.example.mike.mp3player.client.callbacks;

import androidx.viewpager2.widget.ViewPager2;

import com.example.mike.mp3player.client.MediaControllerAdapter;

public class TrackViewPagerChangeCallback extends ViewPager2.OnPageChangeCallback {

    /** */
    private final MediaControllerAdapter mediaControllerAdapter;

    private int currentPosition;

    /** Constructor */
    public TrackViewPagerChangeCallback(MediaControllerAdapter mediaControllerAdapter) {
        this.mediaControllerAdapter = mediaControllerAdapter;
        this.currentPosition = mediaControllerAdapter.getCurrentQueuePosition();
    }
    /**
     *
     * @param position
     */
    @Override
    public void onPageSelected(int position) {

        if (this.currentPosition == position) {
            return;
        }
        if (isSkipToNext(position)) {
            mediaControllerAdapter.skipToNext();
        }
        else if (isSkipToPrevious(position)) {
            mediaControllerAdapter.seekTo(0);
            mediaControllerAdapter.skipToPrevious();
        }

        this.currentPosition = position;
    }

    private boolean isSkipToNext(int position) {
        return position == (currentPosition + 1);
    }

    private boolean isSkipToPrevious(int position) {
        return position == (currentPosition - 1);
    }

    /** */
    public void setCurrentPosition(int currentPosition) {
        this.currentPosition = currentPosition;
    }

    public int getCurrentPosition() {
        return currentPosition;
    }
}

File: app/src/main/java/com/github/goldy1992/mp3player/client/callbacks/TrackViewPagerChangeCallback.kt
package com.github.goldy1992.mp3player.client.callbacks

import androidx.viewpager2.widget.ViewPager2.OnPageChangeCallback
import com.github.goldy1992.mp3player.client.MediaControllerAdapter

class TrackViewPagerChangeCallback(
        /**  */
        private val mediaControllerAdapter: MediaControllerAdapter) : OnPageChangeCallback() {
    /**  */
    var currentPosition: Int

    /**
     *
     * @param position
     */
    override fun onPageSelected(position: Int) {
        if (currentPosition == position) {
            return
        }
        if (isSkipToNext(position)) {
            mediaControllerAdapter.skipToNext()
        } else if (isSkipToPrevious(position)) {
            mediaControllerAdapter.seekTo(0)
            mediaControllerAdapter.skipToPrevious()
        }
        currentPosition = position
    }

    private fun isSkipToNext(position: Int): Boolean {
        return position == currentPosition + 1
    }

    private fun isSkipToPrevious(position: Int): Boolean {
        return position == currentPosition - 1
    }

    /** Constructor  */
    init {
        currentPosition = mediaControllerAdapter.currentQueuePosition
    }
}